CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -182,21 +236,65 @@ void SiPixelLorentzAnglePCLHarvester::dqmEndJob(DQMStore::IBooker& iBooker, DQMS
             << "\t"
             << "chi2"
             << "\t"
-            << "prob" << std::endl;
+            << "prob"
+            << "\t"
+            << "newDetId" << std::endl;
 
   SiPixelLorentzAngle* LorentzAngle = new SiPixelLorentzAngle();
 
+  for (int j = 0; j < (int)hists.newDetIds_.size(); j++) {
+    int new_index = j + 1 + hists.nModules_[hists.nlay - 1] + (hists.nlay - 1) * hists.nModules_[hists.nlay - 1];
+    for (int i = 1; i <= hist_depth_; i++) {
+      findMean(h_drift_depth_adc_slice_, i, new_index);
+    }
+    TF1* f1 = new TF1("f1", "[0] + [1]*x + [2]*x*x + [3]*x*x*x + [4]*x*x*x*x + [5]*x*x*x*x*x", 5., 280.);
```

Comment:
The code should run faster if you allocate `f1` on stack `TF1 f1 = TF1(...)`. Even better, why not allocate in only once and then use for all the modules. Before each fit you would just need to reset parameter values to their initial values.

Reply:
And yes, if allocated on stack, all the function call that use pointers will have to take `&f1` as input.